File: backend/app/api/v1/health_routes.py
import platform
import sys

import fastapi
import uvicorn
from fastapi import APIRouter

from app.services.prompt_classifier_service import prompt_classifier

router = APIRouter()


@router.get("/health")
def get_health():
    """
    Health check endpoint returning application status, ML model availability
    status, and the backend runtime versions (surfaced in the Electron
    DevTools console banner alongside the Electron/Chromium versions).
    """
    return {
        "status": "healthy",
        "version": "1.0.0",
        "model_loaded": prompt_classifier.model_loaded,
        "classifier_mode": prompt_classifier.classifier_mode,
        "runtime": {
            "python": platform.python_version(),
            "python_implementation": f"{platform.python_implementation()} ({sys.version.split()[0]})",
            "fastapi": fastapi.__version__,
            "uvicorn": uvicorn.__version__,
            "platform": platform.platform(),
        },
    }

File: backend-node/src/routes/health.routes.ts
import type { FastifyInstance } from 'fastify'
import os from 'node:os'
import process from 'node:process'
import { promptClassifier } from '../services/promptClassifierService.js'

// Mirrors health_routes.py: status/version plus the live classifier mode and the
// backend runtime versions surfaced in the Electron DevTools console banner.
export default async function healthRoutes(app: FastifyInstance): Promise<void> {
  app.get('/health', async () => ({
    status: 'healthy',
    version: '1.0.0',
    model_loaded: promptClassifier.modelLoaded,
    classifier_mode: promptClassifier.classifierMode,
    // 'fp32' whenever the DL model is active: the unquantized graph is a
    // deliberate accuracy-over-speed choice and should be visible externally.
    model_precision: promptClassifier.modelPrecision,
    runtime: {
      node: process.version,
      node_implementation: `Node.js (${process.version})`,
      fastify: app.version,
      platform: `${os.type()} ${os.release()} (${os.arch()})`,
    },
  }))
}
